eth/backend.go

```diff
@@ -552,7 +558,15 @@ func (s *Ethereum) IsListening() bool                  { return true } // Always
 func (s *Ethereum) EthVersion() int                    { return int(s.protocolManager.SubProtocols[0].Version) }
 func (s *Ethereum) NetVersion() uint64                 { return s.networkID }
 func (s *Ethereum) Downloader() *downloader.Downloader { return s.protocolManager.downloader }
-func (s *Ethereum) GasFeeRecipient() common.Address    { return s.config.Etherbase }
+func (s *Ethereum) GatewayFee() *big.Int               { return s.gatewayFee }
+
+func (s *Ethereum) GatewayFeeRecipient() common.Address {
+	etherbase, err := s.Etherbase()
```

Comment:
Is it more efficient for us to return `nil` here? That way we don't need to actually process the gateway fee

Reply:
I don't quite understand this comment? What do you mean by "we don't actually need to process the gateway fee"?

Reply:
Ah I understand now. This is the full-node api code and a full-node has no reason to specify a gateway fee. I had misunderstood what this package was used for, but found the error just now while working on the `transfer_test.ts`